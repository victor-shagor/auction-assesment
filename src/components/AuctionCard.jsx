import * as React from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Button from "@mui/material/Button";
import Box from "@mui/material/Box";
import Avatar from "@mui/material/Avatar";
import Stack from "@mui/material/Stack";
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles({
  card: {
    paddingTop: "10px",
    border: "1px solid #D0D5DD",
    borderRadius: "12px",
    fontFamily: "IBMPlexRegular !important"
  },
  box: {
    borderRadius: "20px",
    width: "100%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    heigth: "200px",
  },
  avatar: {
    backgroundColor: "#F2F4F7 !important",
    color: "#475467 !important",
    fontSize: '14px !important',
    fontFamily: "IBMPlexRegular !important"
  },
  btn: {
    width: "100%",
    backgroundColor: "#004CCC !important",
    borderRadius: "8px !important",
    textTransform: "none !important",
    fontFamily: "IBMPlexRegular !important"
  },
  title:{
    fontFamily: "IBMPlexSemiBold"
  },
  name:{
    fontFamily: "IBMPlexSemiBold",
    fontSize: '12px'
  },
  nameSpan:{
    color: '#98A2B3',
    fontFamily: "IBMPlexRegular"
  },
  bid:{
    color: '#344054'
  }
});

const AuctionCard = ({ product }) => {
  const classes = useStyles();
  const { name, title, bid, image } = product
  const initials = name
    ?.split(" ")
    .map((n) => n[0])
    .join(".");
  return (
    <Card elevation={0} className={classes.card} >
      <Box className={classes.box}>
        <img src={image} alt="" style={{width: '95%'}}/>
      </Box>
      <CardContent>
        <Stack
          direction="row"
          alignItems="center"
          spacing={1}
          paddingBottom="5px"
        >
          <Avatar sx={{ width: 24, height: 24 }} className={classes.avatar}>{initials}</Avatar>
          <p className={classes.name}>{name} <span className={classes.nameSpan}>(Highest Bidder)</span></p>
        </Stack>
        <p className={classes.title}>
          {title}
        </p>
        <p className={classes.bid}>
          Current Bid: <span className={classes.title}>{bid}</span>
        </p>
        <Button variant="contained" disableElevation className={classes.btn}>
          Add to wishlist
        </Button>
      </CardContent>
    </Card>
  );
};

export default AuctionCard;
