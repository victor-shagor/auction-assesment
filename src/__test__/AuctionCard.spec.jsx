import { render, screen } from "@testing-library/react";
import AuctionCard from "../components/AuctionCard";

describe("Auction card component", () => {
  it("should render auction card", async () => {
    render(
      <AuctionCard
        product={{ name: "test name", bid: "N100", title: "test title" }}
      />
    );

    expect(screen.getByTestId("auction-card")).toBeInTheDocument();
    expect(screen.getByTestId("auction-title").innerHTML).toBe("test title");
  });

  it("should show correct initals in avatar", async () => {
    render(
      <AuctionCard
        product={{ name: "test name", bid: "N100", title: "test title" }}
      />
    );

    expect(screen.getByTestId("name-avatar").innerHTML).toBe("tn");
  });

  it("renders correctly", () => {
    const { container } = render(
      <AuctionCard product={{ title: "test title" }} />
    );
    expect(container).toMatchSnapshot();
  });
});
