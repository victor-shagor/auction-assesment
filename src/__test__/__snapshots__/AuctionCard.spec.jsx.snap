// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auction card component renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root makeStyles-card-17 css-xg2nla-MuiPaper-root-MuiCard-root"
    data-testid="auction-card"
  >
    <div
      class="makeStyles-box-18 MuiBox-root css-0"
    >
      <img
        alt=""
        width="95%"
      />
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiStack-root css-1kmvrgo-MuiStack-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault makeStyles-avatar-19 css-1f0tmi6-MuiAvatar-root"
          data-testid="name-avatar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <p
          class="makeStyles-name-22"
        >
           
          <span
            class="makeStyles-nameSpan-23"
          >
            (Highest Bidder)
          </span>
        </p>
      </div>
      <p
        class="makeStyles-title-21"
        data-testid="auction-title"
      >
        test title
      </p>
      <p
        class="makeStyles-bid-24"
      >
        Current Bid: 
        <span
          class="makeStyles-title-21"
        />
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation makeStyles-btn-20 css-zddlty-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add to wishlist
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
